applications/FluidDynamicsApplication/custom_processes/embedded_skin_visualization_process.cpp

```diff
@@ -128,14 +128,28 @@ void EmbeddedSkinVisualizationProcess::ExecuteBeforeSolutionLoop() {
 
     if (mSetVisualizationMesh){
 
-        const unsigned int n_nodes = mrModelPart.NumberOfNodes();
-        const unsigned int n_elems = mrModelPart.NumberOfElements();
-        const unsigned int n_conds = mrModelPart.NumberOfConditions();
+        int n_nodes = mrModelPart.NumberOfNodes();
+        int n_elems = mrModelPart.NumberOfElements();
+        int n_conds = mrModelPart.NumberOfConditions();
+
+        int n_nodes_scansum, n_elems_scansum, n_conds_scansum;
+        mrModelPart.GetCommunicator().ScanSum(n_nodes, n_nodes_scansum);
+        mrModelPart.GetCommunicator().ScanSum(n_elems, n_elems_scansum);
+        mrModelPart.GetCommunicator().ScanSum(n_conds, n_conds_scansum);
+
+        int n_nodes_sumall(n_nodes);
+        int n_elems_sumall(n_elems);
+        int n_conds_sumall(n_conds);
+        mrModelPart.GetCommunicator().SumAll(n_nodes_sumall);
+        mrModelPart.GetCommunicator().SumAll(n_elems_sumall);
+        mrModelPart.GetCommunicator().SumAll(n_conds_sumall);
```

Comment:
I do not understand this part. The scansum gives you the cumulative numbers of existing elements, which does not give you the guarantee that you will not repeat Ids (think of the case where you create more nodes than there are now in the modelpart). To ensure uniqueness of Ids you need to calculate the scansum of nodes to be added, so that the ids at rank `j+1` start as `num_existing_nodes + num_of_added_nodes_in_ranks_0_to_j + 1`

Reply:
This is just the temporal id. What you say is done at the end of the method (the added nodes, elements and conditions ids. are reassigned to be consecutive).